src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/IdentifierOfTypeSearchValue.cs

```diff
@@ -0,0 +1,109 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using EnsureThat;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.SearchValues
+{
+    /// <summary>
+    /// Represents a identifer with :of-type modifier search value.
+    /// </summary>
+    public class IdentifierOfTypeSearchValue : ISearchValue
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IdentifierOfType"/> class.
+        /// </summary>
+        /// <param name="system">The identifier system value.</param>
+        /// <param name="code">The identifier code value.</param>
+        /// <param name="value">The identifier value.</param>
+        public IdentifierOfTypeSearchValue(string system, string code, string value)
+        {
+            // All values must be present and non empty.
+            EnsureArg.IsTrue(
+                !string.IsNullOrWhiteSpace(system) &&
+                !string.IsNullOrWhiteSpace(code) &&
+                !string.IsNullOrWhiteSpace(value));
```

Comment:
For consistency, EnsureArg.IsNotNullOrWhiteSpace(value, name) would be the same check, but would also expose which parameter broke the contract.

```suggestion
            EnsureArg.IsNotNullOrWhiteSpace(system, nameof(system));
            EnsureArg.IsNotNullOrWhiteSpace(code, nameof(code));
            EnsureArg.IsNotNullOrWhiteSpace(value, nameof(value));
```